src/Compilers/CSharp/Portable/Symbols/TupleTypeSymbol.cs

```diff
@@ -51,46 +52,121 @@ internal sealed class TupleTypeSymbol : NamedTypeSymbol, ITupleTypeSymbol
             out ImmutableArray<TupleFieldSymbol> fields
             )
         {
-            NamedTypeSymbol underlyingType;
-            fields = ImmutableArray<TupleFieldSymbol>.Empty;
+            Debug.Assert(elementNames.IsDefaultOrEmpty || elementTypes.Length == elementNames.Length);
+            Debug.Assert(elementTypes.Length > 1 && elementTypes.Length < 8); // PROTOTYPE
 
-            switch (elementTypes.Length)
+            var tupleType = binder.GetWellKnownType(EnumType(elementTypes.Length), diagnostics, syntax);
+            NamedTypeSymbol underlyingType = tupleType.Construct(elementTypes);
+
+            if (elementTypes.Length > 1 && elementTypes.Length < 8)
```

Comment:
I am not sure what is the purpose of this ```if```, it looks like we are going to throw in ```EnumType(elementTypes.Length)``` above.